### ISSUE TITLE
[BUG] Actions are failing with 'image not known' 

### ISSUE THREAD

**Issue by etsauer** [redhat-actions/push-to-registry#42]
### Version

redhat-actions/push-to-registry@v2

### Describe the bug

When attempting to push, the action fails with error message like this:

Pushing "docker.io/library/committime-exporter" with tag "280f5dcf5d8ddc945923f8141603046d534d0f99" to "quay.io/pelorus" as "***"
Combining image name "committime-exporter" and registry "quay.io/pelorus" to form registry path "quay.io/pelorus/committime-exporter"
/usr/bin/podman push --quiet --digestfile docker.io-library-committime-exporter-280f5dcf5d8ddc945923f8141603046d534d0f99_digest.txt docker.io/library/committime-exporter:280f5dcf5d8ddc945923f8141603046d534d0f99 quay.io/pelorus/committime-exporter:280f5dcf5d8ddc945923f8141603046d534d0f99 --tls-verify=true --creds=***:***
Error: docker.io/library/committime-exporter:280f5dcf5d8ddc945923f8141603046d534d0f99: image not known
Error: podman exited with code 125
Error: docker.io/library/committime-exporter:280f5dcf5d8ddc945923f8141603046d534d0f99: image not known

It look like the same thing is happening in this repository as well.

### Steps to reproduce, workflow links, screenshots

See https://github.com/redhat-actions/push-to-registry/actions/runs/1035845452

**Comment by Divyansh42** [comment]
This issue is probably happening after the Github actions virtual environment picked up Podman version `3.3.2`. When you pull the image from the local docker daemon. By default, `localhost` gets added in front of the image name that got pulled.

Example:
| Image name in Docker daemon | Image name after pulling it to Podman
| ------ | --------- |
| myimage | localhost/myimage |
| namespace/myimage | localhost/namespace/myimage |
| quay.io/namespace/myimage | localhost/quay.io/namespace/myimage | 

Earlier behavior was very different.
Example:
| Image name in Docker daemon | Image Name after pulling it to Podman
| ------ | --------- |
| myimage | docker.io/library/myimage |
| namespace/myimage | docker.io/namespace/myimage |
| quay.io/namespace/myimage | quay.io/namespace/myimage | 

We will update this action based on the above changes as soon as possible.

**Comment by Divyansh42** [comment]
As we are finding the best possible solution.
Meanwhile, you can use the Ubuntu version `ubuntu-18.04` that is still using Podman version `3.0.1`.
Reference: https://github.com/actions/virtual-environments/blob/main/images/linux/Ubuntu1804-README.md

Probably this will fix this issue temporarily,

**Comment by tetchel** [comment]
The 20.04 podman version jumped from 3.1.2 to 3.2.2 [ref](https://github.com/actions/virtual-environments/commit/0c464086ae2bed1df7411a41bab12b674bc421e0#diff-6be9f9ce03ba57ca01e4ae937d5c2bb04893520fdf391ea87eb46acbc43ecdc5R89)
So, the change was introduced somewhere between those two versions.

I do not see any obvious reference to this change in the podman release notes, but we have to figure out exactly which version changed this behaviour, so we can handle it in the right cases.

**Comment by Divyansh42** [comment]
Yes, I also tried to find out that but there is nothing in the release docs of all the versions after 3.1.2 and there doesn't seem to be a way to install older versions of the podman to check this out manually.

**Comment by tetchel** [comment]
The podman team is treating this as a bug (see linked issue) so we will wait for a fix on their end. Again, you may use ubuntu-18.04 in the meantime.

We are adding a test case which pulls the newest podman version so hopefully we can catch any upcoming issues before podman is updated in the GitHub runners.

**Comment by andmos** [comment]
FYI, I think (?) I see this error with podman 3.2.3 also,
```
Tags "2.0.6, ab37703" of "ip-avgivelser" found in Docker image storage
"docker.io/library/ip-avgivelser" was found in the Docker image storage, but not in the Podman image storage. The image(s) will be pulled into Podman image storage, pushed, and then removed from the Podman image storage.
⏳ Pushing "docker.io/library/ip-avgivelser" with tags "2.0.6, ab37703" to "ghcr.io/domstolene" as "domstolene"
Combining image name "ip-avgivelser" and registry "ghcr.io/domstolene" to form registry path "ghcr.io/domstolene/ip-avgivelser"
/usr/bin/podman push --quiet --digestfile docker.io-library-ip-avgivelser-2.0.6_digest.txt docker.io/library/ip-avgivelser:2.0.6 ghcr.io/domstolene/ip-avgivelser:2.0.6 --tls-verify=true --creds=domstolene:***
Error: docker.io/library/ip-avgivelser:2.0.6: image not known
Error: podman exited with code 125
Error: docker.io/library/ip-avgivelser:2.0.6: image not known
```

**Comment by Divyansh42** [comment]
Podman `3.2.3` is released 18 days ago and they have fixed this issue after this release, since then they haven't done any release.

**Comment by tetchel** [comment]
Podman [3.3.1](https://github.com/containers/podman/releases/tag/v3.3.1) should resolve this but it has not been picked up by GitHub runners yet.

**Comment by divyansh42** [comment]
`Ubuntu 20.04` is updated with podman version `3.3.1`.
This update should fix this issue.
